integration-test/1555-be-more-robust-to-none.py

```diff
@@ -36,3 +36,29 @@ def test_fake_rel_with_ref_none(self):
                 'id': 367163748,
                 'network': 'PL:motorway',
             })
+
+    def test_es_ref_no_digits(self):
+        import dsl
+        z, x, y = (16, 32360, 24673)
+        self.generate_fixtures(
+            dsl.is_in('ES', z, x, y),
+            dsl.way(1, dsl.tile_diagonal(z, x, y), {
+                'highway': 'motorway',
+                'source': 'openstreetmap.org',
+            }),
+            dsl.relation(1, {
+                # NOTE: no "ref" on the relation.
+                'network': 'ES:A-road',
+                'route': 'road',
+                'type': 'route',
+                'source': 'openstreetmap.org',
+            }, ways=[1]),
+        )
+
+        # can't figure out network or shield_text for this!
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 1,
+                'network': type(None),
```

Comment:
It's kinda weird to have `type(None)` here rather than just `None`.

Reply:
Yup, it's totally weird. It's to work around the previous meaning of `None` as "anything", while abusing the type matching to match `None` because it's the only value of `NoneType`. I've added #1564 to follow up - hopefully we can make the "wildcards" in the property matching clearer.